app/models/comment.rb

```diff
@@ -0,0 +1,33 @@
+# Model class for comments
+class Comment < ApplicationRecord
+  belongs_to :discussion
+  belongs_to :parent, class_name: 'Comment', optional: true
+
+  has_many :comments, inverse_of: :parent,
+                      foreign_key: :parent_id,
+                      dependent: :destroy
+
+  validates :discussion_id, presence: true
+  validates :number, uniqueness: { scope: :discussion_id }
+
+  validates :body,  presence: true, unless: -> { title.present? }
+  validates :title, presence: true, unless: -> { body.present?  }
+
+  # Instance methods
+
+  def lock
```

Comment:
There *might* be an existing `lock` instance method provided by ActiveRecord. There is a `lock!` one for sure, so this naming is a little risky.